parcels/particle.py

```diff
@@ -148,6 +176,9 @@ def cptr(i):
             lon = np.linspace(start[0], finish[0], size, dtype=np.float32)
             lat = np.linspace(start[1], finish[1], size, dtype=np.float32)
 
+        if start_field is not None:
+            lon, lat = positions_from_density_field(size, getattr(self.grid, start_field))
```

Comment:
Could we not simply pass in the `Field` object itself instead of the name? We could even do some type checking at this point.